desktop/main/NodeConfig.ts

```diff
@@ -21,19 +24,96 @@ export const loadNodeConfig = async (): Promise<NodeConfig> =>
         )
     : {};
 
-const loadSmeshingOpts = async (nodeConfig) => {
-  const id = generateGenesisIDFromConfig(nodeConfig);
-  const opts = StoreService.get(`smeshing.${id}`);
-  return safeSmeshingOpts(opts, id);
+const getCustomNodeConfigName = (name: string) => `node-config.${name}.json`;
+
+const loadCustomNodeConfig = async (
+  path: string,
+  defaultValue = () => ({})
+): Promise<Partial<NodeConfig>> =>
+  existsSync(path)
+    ? fs
+        .readFile(path, {
+          encoding: 'utf8',
+        })
+        .then((res) => JSON.parse(res))
+    : defaultValue();
+
+export const updateSmeshingOptsInCustomConfig = async (
+  netName: string,
+  opts: Partial<NodeConfig['smeshing']>
+): Promise<Partial<NodeConfig>> => {
+  const netConfigPath = path.join(
+    USERDATA_DIR,
+    getCustomNodeConfigName(netName)
+  );
+  const customConfig = await loadCustomNodeConfig(netConfigPath);
+
+  customConfig.smeshing = opts;
+
+  await fs.writeFile(netConfigPath, JSON.stringify(customConfig, null, 2), {
+    encoding: 'utf8',
+  });
+
+  return customConfig;
+};
+
+const migrateStoreSemshingOpts = (genesisID: string) => {
+  const opts = StoreService.get(`smeshing.${genesisID}`);
+  StoreService.remove(`smeshing.${genesisID}`);
+  if (opts) {
+    return opts;
+  }
+
+  return {};
+};
+
+const isCustomNodeConfigExists = (netName: string) => {
+  const configPath = path.join(USERDATA_DIR, getCustomNodeConfigName(netName));
+  return existsSync(configPath);
+};
+
+export const loadOrCreateCustomConfig = async (
+  netName: string,
+  genesisID?: string
+): Promise<Partial<NodeConfig>> => {
+  const configPath = path.join(USERDATA_DIR, getCustomNodeConfigName(netName));
+  let smeshingOpts = {};
+
+  // migrate opts or set default
+  if (genesisID) {
+    const migrateOptsConfig = migrateStoreSemshingOpts(genesisID);
+    // migrate options from StoreService or place only default opts
+    smeshingOpts = !R.isEmpty(migrateOptsConfig)
+      ? migrateOptsConfig
+      : safeSmeshingOpts(undefined, genesisID); // set default dir path
+  }
+
+  return loadCustomNodeConfig(configPath, () =>
+    updateSmeshingOptsInCustomConfig(netName, smeshingOpts)
+  );
 };
 
-export const downloadNodeConfig = async (networkConfigUrl: string) => {
-  const nodeConfig = await fetchNodeConfig(networkConfigUrl);
-  // Copy smeshing opts from previous node config or replace it with empty one
-  nodeConfig.smeshing = await loadSmeshingOpts(nodeConfig);
+export const downloadNodeConfig = async (
+  netName: string,
+  networkConfigUrl: string
+) => {
+  // load configs
+  const discoveryConfig = await fetchNodeConfig(networkConfigUrl);
+  const clientConfig = await loadNodeConfig();
+
+  // do not update config file if exists, and forward to $nodeConfig.next
+  // check regression for new apps version which does not have custom config
+  if (!R.isEmpty(clientConfig) && isCustomNodeConfigExists(netName)) {
+    return discoveryConfig;
+  }
```

Comment:
1. I don't understand the purpose of this. However it looks very misleading, like `downloadNodeConfig` may return some valid NodeConfig, but does not "download" (in the meaning of fetching and saving on fs) it actually
2. And it does not seem correct at all. Even if we have config and custom config for the network — why we don't merge them?
3. And finally, keep in mind that we may switch networks / unlock another wallet (with another network), etc. In other words, there might be presented node-config.json valid for another network, while User may have a custom config for the target network (aka User switches between testnet and mainnet from time to time).